dbt/parser.py

```diff
@@ -545,12 +551,10 @@ def parse_schema_test(test_base, model_name, test_config, test_type,
     # sort the dict so the keys are rendered deterministically (for tests)
     kwargs = [as_kwarg(key, test_args[key]) for key in sorted(test_args)]
 
-    macro_name = "test_{}".format(test_type)
-
-    if package_project_config.get('name') != \
-       root_project_config.get('name'):
-        macro_name = "{}.{}".format(package_project_config.get('name'),
-                                    macro_name)
+    if test_namespace is None:
+        macro_name = "test_{}".format(test_type)
+    else:
+        macro_name = "{}.test_{}".format(test_namespace, test_type)
```

Comment:
this is totally reasonable.

taking a higher view: i don't like that the path has to be manipulated like this to inject `test_`, i think this would all be a lot cleaner if we just called a macro by its name instead of by its name, plus `test_`. then we don't have to do this manipulation, we just pass the name through.